### PR TITLE
build: use join_paths()

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

File: src/tests/meson.build [hunk 1]
```diff
@@ -22,7 +22,7 @@ unit_tests = [
 foreach unit: unit_tests
   exe = executable(unit, unit + '.c',
                    install: true,
-                   install_dir: 'share/installed-tests/graphene-1.0',
+                   install_dir: join_paths(graphene_datadirm, 'installed-tests', 'graphene-1.0'),
```
Comment (ebassi) [inline src/tests/meson.build]: Typo, here: `graphene_datadir` has a trailing `m`.

**Comment by ignatenkobrain** [comment]
@ebassi Fixed!

File: src/tests/meson.build (hunk 1 above)
Comment (ignatenkobrain) [inline src/tests/meson.build]: need new old-style thinkpad keyboard to not type `m` along with `,` ;)

**Comment by ignatenkobrain** [comment]
@ebassi I also fixed libdir abspath, because pkgconfig had relative.

File: doc/meson.build [hunk 2]
```diff
@@ -35,9 +35,9 @@ gnome.gtkdoc('graphene',
                '--ignore-headers=' + ' '.join(private_headers),
              ],
              fixxref_args: [
-               '--html-dir=@0@/@1@/gtk-doc/html'.format(get_option('prefix'), get_option('datadir')),
-               extra_dir_arg + '/glib',
-               extra_dir_arg + '/gobject',
+               '--html-dir=@0@'.format(docpath),
+               '--extra-dir=@0@'.format(join_paths(docpath, 'glib'),
```
Comment (ebassi) [inline doc/meson.build]: Missing closing ')'

File: doc/meson.build [hunk 3]
```diff
@@ -35,9 +35,9 @@ gnome.gtkdoc('graphene',
...
+               '--extra-dir=@0@'.format(join_paths(docpath, 'gobject'),
```
Comment (ebassi) [inline doc/meson.build]: Missing closing ')'

File: meson.build [hunk 4]
```diff
@@ -34,19 +34,9 @@ revision = graphene_interface_age
...
-else
-  graphene_includedir = get_option('includedir')
-endif
-
-if get_option('datadir') == 'share'
-  graphene_datadir = graphene_prefix + '/share'
-else
-  graphene_datadir = get_option('datadir')
-endif
+graphene_libdir = join_path(graphene_prefix, get_option('libdir'))
```
Comment (ebassi) [inline meson.build]: Typo: should be `join_paths`.

**Comment by ignatenkobrain** [comment]
argh, fixed 3 errors from review

**Pull request merged**